pkg/operator/baremetal_test.go

```diff
@@ -341,3 +344,159 @@ func TestBaremetalProvisionigConfig(t *testing.T) {
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "metal3",
+					Annotations: map[string]string{
+						cboOwnedAnnotation: "",
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			testCase: "Both cboOwnedAnnotation and maoOwnedAnnotation",
+			deployment: &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Deployment",
+					APIVersion: "apps/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "metal3",
+					Annotations: map[string]string{
+						cboOwnedAnnotation: "",
+						maoOwnedAnnotation: "",
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			testCase: "No cboOwnedAnnotation or maoOwnedAnnotation",
+			deployment: &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Deployment",
+					APIVersion: "apps/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "metal3",
+					Annotations: map[string]string{},
+				},
+			},
+			expected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(string(tc.testCase), func(t *testing.T) {
+
+			_, err := client.Deployments("test-namespace").Create(context.Background(), tc.deployment, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Could not create metal3 test deployment.\n")
+			}
+			maoOwned, err := checkMetal3DeploymentMAOOwned(client, operatorConfig)
+			if maoOwned != tc.expected {
+				t.Errorf("Expected: %v, got: %v", tc.expected, maoOwned)
+			}
+			if tc.expectedError != (err != nil) {
+				t.Errorf("ExpectedError: %v, got: %v", tc.expectedError, err)
+			}
+			err = client.Deployments("test-namespace").Delete(context.Background(), "metal3", metav1.DeleteOptions{})
+			if err != nil {
+				t.Errorf("Could not delete metal3 test deployment.\n")
+			}
+		})
+	}
+
+}
+
+func TestCheckForBaremetalClusterOperator(t *testing.T) {
+	testCases := []struct {
+		testCase        string
+		clusterOperator *osconfigv1.ClusterOperator
+		expected        bool
+		expectedError   bool
+	}{
+		{
+			testCase: "CO Baremetal",
+			clusterOperator: &osconfigv1.ClusterOperator{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ClusterOperator",
+					APIVersion: "config.openshift.io/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: cboClusterOperatorName,
+				},
+				Status: osconfigv1.ClusterOperatorStatus{
+					RelatedObjects: []osconfigv1.ObjectReference{
+						{
+							Group:    "",
+							Resource: "namespaces",
+							Name:     "openshift-machine-api",
+						},
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			testCase: "CO Test",
+			clusterOperator: &osconfigv1.ClusterOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+			},
+			expected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(string(tc.testCase), func(t *testing.T) {
+			var osClient *fakeos.Clientset
+			osClient = fakeos.NewSimpleClientset(tc.clusterOperator)
+			_, err := osClient.ConfigV1().ClusterOperators().Create(context.Background(), tc.clusterOperator, metav1.CreateOptions{})
```

Comment:
This unit test is unable to create the ClusterOperator for the CBO using this fake client. Please let me know if I am missing something here.

Reply:
Fixed it.